### PR TITLE
Add metric provider

### PR DESCRIPTION
**Pull request by pfi79**

### CODE REVIEW

File: test/basic_test.go [hunk 1]
```diff
@@ -1630,35 +1630,33 @@ func TestMigrateToBlacklistAndBackAgain(t *testing.T) {
...
+	// 	assert.Equal(t, uint64(2), atomic.LoadUint64(&boundCommitSignatures))
+	//
+	// 	nodes[0].Submit(Request{ID: "6", ClientID: "alice"})
+	//
+	// 	for i := 0; i < numberOfNodes; i++ {
+	// 		<-nodes[i].Delivered
+	// 	}
+	//
+	// 	assert.Equal(t, uint64(2), atomic.LoadUint64(&boundCommitSignatures))
+	// })
```
Comment (C0rWin) [inline test/basic_test.go]: You can to t.Skip() instead of commenting out.

Comment (pfi79) [reply test/basic_test.go]: fix

File: pkg/wal/writeaheadlog.go [hunk 2]
```diff
@@ -129,7 +132,15 @@ func Create(logger api.Logger, dirPath string, options *Options) (*WriteAheadLog
...
 	if options != nil {
-		opt = options
+		if options.MetricsProvider != nil {
+			opt.MetricsProvider = options.MetricsProvider
+		}
+		if options.FileSizeBytes != 0 {
+			opt.FileSizeBytes = options.FileSizeBytes
+		}
+		if options.BufferSizeBytes != 0 {
+			opt.BufferSizeBytes = options.FileSizeBytes
```
Comment (C0rWin) [inline pkg/wal/writeaheadlog.go]: FileSizeBytes? or BufferSizeBytes?

File: pkg/wal/writeaheadlog.go [hunk 3]
```diff
@@ -207,7 +218,15 @@ func Open(logger api.Logger, dirPath string, options *Options) (*WriteAheadLogFi
...
```
Comment (C0rWin) [inline pkg/wal/writeaheadlog.go]: FileSizeBytes? or BufferSizeBytes?

File: pkg/wal/writeaheadlog.go (hunk 2 above)
Comment (pfi79) [reply pkg/wal/writeaheadlog.go]: fix
BufferSizeBytes

File: pkg/wal/writeaheadlog.go (hunk 3 above)
Comment (pfi79) [reply pkg/wal/writeaheadlog.go]: fix
BufferSizeBytes

**Review by C0rWin — APPROVED** [review]

**Pull request merged**